src/main/java/teammates/storage/search/InstructorSearchDocument.java

```diff
@@ -96,7 +97,7 @@ public static InstructorSearchResultBundle fromResults(Results<ScoredDocument> r
     private static void sortInstructorResultList(List<InstructorAttributes> instructorList) {
 
         instructorList.sort(Comparator.comparing((InstructorAttributes instructor) -> instructor.courseId)
-                .thenComparing(instructor -> instructor.role)
+                .thenComparing(instructor -> SanitizationHelper.desanitizeIfHtmlSanitized(instructor.role))
```

Comment:
Add `//TODO` to say that you are going to remove this in the future after data migration.